Mathlib/Topology/Algebra/NormedValued.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.RingTheory.Valuation.RankOne
+import Mathlib.Topology.Algebra.Valuation
+
+
+/-!
+# Correspondence between nontrivial nonarchimedean norms and rank one valuations
+
+Nontrivial nonarchimedean norms correspond to rank one valuations.
+
+## Main Definitions
+* `NormedField.toValued` : the valued field structure on a nonarchimedean normed field `K`,
+  determined by the norm.
+* `Valued.toNormedField` : the normed field structure determined by a rank one valuation.
+
+## Tags
+
+norm, nonarchimedean, nontrivial, valuation, rank one
+-/
+
+
+noncomputable section
+
+open Filter Set Valuation
+
+open scoped NNReal
+
+variable {K : Type*} [hK : NormedField K] (h : IsNonarchimedean (norm : K → ℝ))
+
+namespace NormedField
+
+/-- The valuation on a nonarchimedean normed field `K` defined as `nnnorm`. -/
+def valuation : Valuation K ℝ≥0 where
+  toFun           := nnnorm
+  map_zero'       := nnnorm_zero
+  map_one'        := nnnorm_one
+  map_mul'        := nnnorm_mul
+  map_add_le_max' := h
+
+theorem valuation_apply (x : K) : valuation h x = ‖x‖₊ := rfl
+
+/-- The valued field structure on a nonarchimedean normed field `K`, determined by the norm. -/
+def toValued : Valued K ℝ≥0 :=
+  { hK.toUniformSpace,
+    @NonUnitalNormedRing.toNormedAddCommGroup K _ with
+    v := valuation h
+    is_topological_valuation := fun U => by
+      rw [Metric.mem_nhds_iff]
+      exact ⟨fun ⟨ε, hε, h⟩  =>
+          ⟨Units.mk0 ⟨ε, le_of_lt hε⟩ (ne_of_gt hε), fun x hx ↦ h (mem_ball_zero_iff.mpr hx)⟩,
+        fun ⟨ε, hε⟩ => ⟨(ε : ℝ), NNReal.coe_pos.mpr (Units.zero_lt _),
+          fun x hx ↦ hε (mem_ball_zero_iff.mp hx)⟩⟩ }
+
+end NormedField
+
+namespace Valued
+
+variable {L : Type*} [Field L] {Γ₀ : Type*} [LinearOrderedCommGroupWithZero Γ₀]
+  [val : Valued L Γ₀] [hv : RankOne val.v]
+
+/-- The norm function determined by a rank one valuation on a field `L`. -/
+def norm_def : L → ℝ := fun x : L => hv.hom (Valued.v x)
```

Comment:
`norm_def` sounds more like the name of a lemma saying that norm = something it's defined to be. Is there a reason that this isn't just called `norm`?

Reply:
I can rename it to `norm`.